staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/controller_reconcile.go

```diff
@@ -91,12 +92,28 @@ func (c *celAdmissionController) reconcilePolicyDefinition(namespace, name strin
 	}
 
 	// find GVR for params
-	paramsGVR, err := c.restMapper.RESTMapping(paramSource.GroupKind(), paramSource.Version)
+	// Parse param source into a GVK
+
+	paramSourceGV, err := schema.ParseGroupVersion(paramSource.APIVersion)
+	if err != nil {
+		// Failed to resolve. Return error so we retry again (rate limited)
+		// Save a record of this definition with an evaluator that unconditionally
+		info.configurationError = fmt.Errorf("failed to parse apiVersion of paramKind '%v' with error: %w", paramSource.String(), err)
+
+		// Return nil, since this error cannot be resolved by waiting more time
+		return nil
+	}
+
+	paramsGVR, err := c.restMapper.RESTMapping(schema.GroupKind{
+		Group: paramSourceGV.Group,
+		Kind:  paramSource.Kind,
+	}, paramSourceGV.Version)
+
 	if err != nil {
 		// Failed to resolve. Return error so we retry again (rate limited)
 		// Save a record of this definition with an evaluator that unconditionally
 		//
-		info.configurationError = fmt.Errorf("failed to find resource for param source: '%v'", paramSource.String())
+		info.configurationError = fmt.Errorf("failed to find resource referenced by paramKind: '%v'", paramSourceGV.WithKind(paramSource.Kind))
 		return info.configurationError
 	}
 
```

Comment:
at some point we decided to allow native types to be used as params. right this code is terribly inefficient use of informers not just for the fact that json is being used but also for the fact that the caches are being duplicated for native types.

I have [a branch](https://github.com/alexzielenski/kubernetes/tree/celIntegration-shared-informer-native-types) tracking this work to be fixed up in a followup PR. When discussing our group decided not to block on this for now, but it will need to be address.

Reply:
Agree it's not a blocker for alpha. I think it should be on the beta graduation criteria list.

Reply:
Added into beta graduation criteria: https://github.com/kubernetes/enhancements/pull/3644